packages/backend/src/models/UserModel.ts

```diff
@@ -1,31 +1,202 @@
 import bcrypt from 'bcrypt';
-import { CreateOrganizationUser } from 'common';
+import {
+    CreateInitialUserArgs,
+    CreateOrganizationUser,
+    LightdashUser,
+    SessionUser,
+    UpdateUserArgs,
+} from 'common';
 import { Knex } from 'knex';
-import { createEmail, EmailTableName } from '../database/entities/emails';
+import {
+    createEmail,
+    deleteEmail,
+    EmailTableName,
+} from '../database/entities/emails';
 import { InviteLinkTableName } from '../database/entities/inviteLinks';
 import { createOrganizationMembership } from '../database/entities/organizationMemberships';
+import { createOrganization } from '../database/entities/organizations';
 import { createPasswordLogin } from '../database/entities/passwordLogins';
 import {
-    createUser,
-    DbUserDetails,
-    getUserDetailsByUuid,
+    DbUser,
+    DbUserIn,
+    DbUserUpdate,
     UserTableName,
 } from '../database/entities/users';
-import { NotExistsError, ParameterError } from '../errors';
+import { NotExistsError, NotFoundError, ParameterError } from '../errors';
 import { InviteLinkModel } from './InviteLinkModel';
 
+export type DbUserDetails = {
+    user_id: number;
+    user_uuid: string;
+    first_name: string;
+    last_name: string;
+    created_at: Date;
+    is_tracking_anonymized: boolean;
+    email: string | undefined;
+    organization_uuid: string;
+    organization_name: string;
+};
 export type DbOrganizationUser = Pick<
     DbUserDetails,
     'user_uuid' | 'first_name' | 'last_name' | 'email'
 >;
 
+export const mapDbUserDetailsToLightdashUser = (
+    user: DbUserDetails,
+): LightdashUser => {
+    if (!user.organization_uuid || !user.organization_name) {
+        throw new NotFoundError(
+            `Cannot find organization for user with uuid ${user.user_uuid}`,
+        );
+    }
+    return {
+        userUuid: user.user_uuid,
+        email: user.email,
+        firstName: user.first_name,
+        lastName: user.last_name,
+        organizationUuid: user.organization_uuid,
+        organizationName: user.organization_name,
+        isTrackingAnonymized: user.is_tracking_anonymized,
+    };
+};
+
+const userDetailsQueryBuilder = (
```

Comment:
@ZeRego - I simplified the query builder by removing the password field, which isn't often needed. This can be extended to add passwords